src/Reader/PHP.php

```diff
@@ -217,5 +220,29 @@ protected function find( $key ) {
 
 		return $this->findNext( $key );
 	}
+
+	public function exists( $key ) {
+		if ( $this->find( strval( $key ) ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	public function firstkey() {
```

Comment:
I see your implementation is basically a transliteration of the [PHP 5.3 implementation](https://github.com/php/php-src/blob/PHP-5.3.29/ext/dba/dba_cdb.c#L245-L284), but you could reduce duplication by having `firstKey()` set `$this->pos = 2048` and then return `$this->nextKey()`.